modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ClientImpl.java

```diff
@@ -870,6 +881,10 @@ private void joinError(IgniteSpiException err) {
                 msg.client(true);
 
                 sockWriter.sendMessage(msg);
+
+                if (log.isDebugEnabled())
+                    log.debug("*** Send heartbeat message from node:    "+msg.creatorNodeId());
+
```

Comment:
Code should not contain redundant newlines.
Check Semantic Units section at https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines for details